src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { InputHTMLAttributes, useEffect, useState } from 'react';
+
+// import './_Slider.scss'
+
+export type Size = 'large' | 'small' | undefined;
+export type Color = 'slider-red' | 'slider-blue';
+
+interface ISliderProps {
+  size?: Size;
+  color?: Color;
+  RangeSlider?: boolean;
+  DoubleRangeSlider?: boolean;
+  minGap?: Number;
+  Tick?: boolean;
+  TickNum?: Number;
+  min?: Number;
+  max?: Number;
+  step?: Number;
+  disabled?: boolean;
+}
+
+export type NativeSliderProps = ISliderProps &
+  InputHTMLAttributes<HTMLInputElement>;
+
+const Slider: React.FC<NativeSliderProps> = (props) => {
+  const {
+    RangeSlider,
+    DoubleRangeSlider,
+    size,
+    color,
+    Tick,
+    TickNum,
+    min,
+    max,
+    step,
+    disabled,
+    ...rest
+  } = props;
+
+  // enable (or disable) the range slider, default is true
+  const enableOrDisable = RangeSlider === false ? RangeSlider : true;
+  // enable double range slider
+  const enableDoubleRange =
+    DoubleRangeSlider === true ? DoubleRangeSlider : false;
+  // max value must bigger than 0, default value is 200
+  const maxValue = (max as number) > 0 ? max : 200;
+  // min value must bigger or equal to 0, and min value must smaller than max, default value is 0
+  const minValue =
+    (min as number) > 0 && (maxValue as number) > (min as number) ? min : 0;
+  // step value must bigger than 0, default value is 1
+  const stepValue = (step as number) > 0 ? step : 1;
+  // tick number must bigger than 0, default value is 5
+  const tickNumber = (TickNum as number) > 0 ? TickNum : 5;
+
+  const [valueRight, setValueRight] = useState((maxValue as number) / 2);
+  const [valueLeft, setValueLeft] = useState((maxValue as number) / 3);

```

Comment:
if there're some simple explanations for these two hooks it would be easier for me to understand. Like why valueRight initiated with max/2 while valueLeft is max/3.